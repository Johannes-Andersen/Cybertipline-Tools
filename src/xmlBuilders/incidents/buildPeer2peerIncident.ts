import { XMLBuilder, XMLParser } from 'fast-xml-parser';
import type { Peer2peerIncident } from '../../types';
import { buildIpCapture } from '../common/buildIpCapture';

const parser = new XMLParser({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  allowBooleanAttributes: true,
});

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildPeer2peerIncident = (
  peer2peerIncident: Peer2peerIncident,
  keyName = 'peer2peerIncident',
): string =>
  builder.build({
    [keyName]: {
      client: peer2peerIncident.client,
      ipCaptureEvent: peer2peerIncident.ipCaptureEvent?.map(
        (e) => parser.parse(buildIpCapture(e)).ipCaptureEvent,
      ),
      fileNames: peer2peerIncident.fileNames,
      additionalInfo: peer2peerIncident.additionalInfo,
    },
  });
