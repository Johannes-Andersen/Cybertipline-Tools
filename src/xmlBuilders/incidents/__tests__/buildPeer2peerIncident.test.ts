import { describe, expect, it } from 'vitest';
import type { Peer2peerIncident } from '../../../types';
import { buildPeer2peerIncident } from '../buildPeer2peerIncident';

describe('buildPeer2peerIncident', () => {
  it('should build a peer to peer incident', () => {
    const peer2peerIncident: Peer2peerIncident = {
      ipCaptureEvent: [
        {
          ipAddress: '192.168.1.1',
        },
        {
          ipAddress: '192.168.1.2',
          possibleProxy: true,
        },
      ],
      client: 'My Client',
      fileNames: 'My File',
      additionalInfo: 'This is a test',
    };

    const result = buildPeer2peerIncident(peer2peerIncident);

    expect(result).toMatchSnapshot();
  });
});
