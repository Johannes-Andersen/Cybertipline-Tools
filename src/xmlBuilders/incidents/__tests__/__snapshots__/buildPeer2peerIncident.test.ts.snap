// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildPeer2peerIncident > should build a peer to peer incident 1`] = `"<peer2peerIncident><client>My Client</client><ipCaptureEvent><ipAddress>192.168.1.1</ipAddress></ipCaptureEvent><ipCaptureEvent><ipAddress>192.168.1.2</ipAddress><possibleProxy>true</possibleProxy></ipCaptureEvent><fileNames>My File</fileNames><additionalInfo>This is a test</additionalInfo></peer2peerIncident>"`;
